src/parser.rs

```diff
@@ -1380,12 +1541,34 @@ fn is_windows_drive_letter(segment: &str) -> bool {
     segment.len() == 2 && starts_with_windows_drive_letter(segment)
 }
 
+/// Wether path starts with a root slash
+/// and a windows drive letter eg: "/c:" or "/a:/"
+fn path_starts_with_windows_drive_letter(s: &str) -> bool {
+    s.len() > 3
+        && matches!(s.as_bytes()[0], b'/' | b'\\' | b'?' | b'#')
+        && starts_with_windows_drive_letter(&s[1..])
+}
+
 fn starts_with_windows_drive_letter(s: &str) -> bool {
-    ascii_alpha(s.as_bytes()[0] as char) && matches!(s.as_bytes()[1], b':' | b'|')
+    ascii_alpha(s.as_bytes()[0] as char)
+        && matches!(s.as_bytes()[1], b':' | b'|')
+        && (s.len() == 2 || matches!(s.as_bytes()[2], b'/' | b'\\' | b'?' | b'#'))
 }
 
+/// https://url.spec.whatwg.org/#start-with-a-windows-drive-letter
 fn starts_with_windows_drive_letter_segment(input: &Input) -> bool {
```

Comment:
This method should probably be calling `starts_with_windows_drive_letter`.

Reply:
That was my intention at first, but I couldnt figure out a way to go from input to a &str, and I decided to back off.
It also means starts_with_windows_drive_letter is only called once, so I'd rather remove it :/